### ISSUE TITLE
Need help with my first script using Oxford API

### ISSUE THREAD

**Issue by veganexe** [hubotio/hubot#1413]
# Description:
	#   Hubot script to fetch definitions.
	#
	# Dependencies:
	#   None
	#
	# Configuration:
	#   HUBOT_OX_AppID: Oxford API App ID
	#   HUBOT_OX_AppKey: Oxford API App Key
	#
	# Commands:
	#   hubot define <word> - Define a given word.

	module.exports = (robot) ->
	  robot.respond /define (.*)/i, (msg) ->
	    APIID = process.env.HUBOT_OX_AppID or null
	    if APIID == null
	      msg.send "HUBOT_OX_AppID environment variable is not provided"
	    APIKEY = process.env.HUBOT_OX_AppKey or null
	    if APIKEY == null
	      msg.send "HUBOT_OX_AppKey environment variable is not provided"
	    msg.http("https://od-api.oxforddictionaries.com/api/v1/entries/en/#{msg.match[1]}/definitions")
	      .headers(Accept: 'application/json') # Pass app_id and app_key headers
	      .get() (err, res, body) ->
		data = JSON.parse(body)
		if data.message
		  msg.send "#{data.message}"
		else
		  msg.send "#{data.definitions}"

Can someone please fix the header part and any other mistakes? :(

**Comment by mose** [comment]
1. you should put an `else` after testing if you have an AppID, so it doesn't execute the rest
2. it's safer to urlencode whatever you pass into an url
3. you can pass multiple headers using an object

for example (for the fun of it, I tested this with a real app_id and a real app_key, and it works fine)
```
module.exports = (robot) ->

  robot.respond /define (.+)/i, (msg) ->
    unless process.env.HUBOT_OX_AppID or process.env.HUBOT_OX_AppKey
      unless process.env.HUBOT_OX_AppID
        msg.send "HUBOT_OX_AppID environment variable is not provided"
      unless process.env.HUBOT_OX_AppKey
        msg.send "HUBOT_OX_AppKey environment variable is not provided"
    else
      term = encodeURIComponent(msg.match[1])
      robot.http("https://od-api.oxforddictionaries.com/api/v1/entries/en/#{term}/definitions")
      .headers(
        Accept: "application/json", 
        app_id: process.env.HUBOT_OX_AppID, 
        app_key: process.env.HUBOT_OX_AppKey)
      .get() (err, res, body) ->
        if err
          msg.send err
        else if res.statusCode isnt 200
          msg.send res.statusCode + ' - ' + body
        else
          data = JSON.parse(body)
          if data.results is 0
            msg.send "No result found"
          else
            if data.results[0].lexicalEntries
              for entry in data.results[0].lexicalEntries
                for entries in entry.entries
                  for sense in entries.senses
                    msg.send "(#{entry.lexicalCategory}) #{sense.definitions[0]}"
            else
              msg.send "No lexical entry found"
```

**Comment by veganexe** [comment]
Wow. Works perfectly. Thanks a ton!

Now I tried to extend that script to work for synonyms and antonyms, by changing the http call, which I'm sure I did correctly, and the msg.send line, which is what I screwed up.

The output shows no error (in hubot log). Even shows the exact number of entries as opposed to curl, even with the lexical category matching (forgot to remove that when I tested)

But can't get the antonym/synonym to return. It outputs like "Noun undefined". Tried sense.antonyms, sense.antonyms.id, and all I could think off. What is wrong?
